### PR TITLE
return a Dependency instance from Blocks.load event listener

### PR DESCRIPTION
**Pull request by anentropic**
# Description

So that you can chain `then` off it like other event methods

Closes: #4302 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by anentropic** [comment]
Trying to work out how to run the tests locally, is there a guide anywhere?

So far I have:
- install `hatch`
- `hatch shell`
- install test deps `pip install -r test/requirements.txt`
- `chmod +x scripts/*.sh`
- `scripts/build_frontend.sh`

that seems to be enough to be able to run `pytest -v test/test_blocks.py`

**Comment by abidlabs** [comment]
Thanks for the PR @anentropic, taking a look. You can run tests locally by following the contributing guidelines: https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md#continuous-integration-and-testing

File: test/test_events.py [hunk 1]
```diff
@@ -69,6 +73,25 @@ def clear():
...
+            out = gr.Textbox(label="Call counter")
+            demo.load(increment, inputs=None, outputs=out).then(
+                increment, inputs=None, outputs=out
+            )
+
+        assert demo.config["dependencies"][0]["trigger"] == "load"
+        assert demo.config["dependencies"][0]["trigger_after"] is None
+        assert demo.config["dependencies"][1]["trigger"] == "then"
+        assert demo.config["dependencies"][1]["trigger_after"] == 0
+        demo.launch(prevent_thread_lock=True)
```
Comment (abidlabs) [inline test/test_events.py:93]: No need to launch, just slows down the test
```suggestion
```

**Comment by abidlabs** [comment]
Thanks @anentropic for the PR, I agree in principle that we should make this change. However, I noticed a bug here, which is that setting the trigger to be the high-level `Blocks` instance causes a problem if that `Blocks` is re-used within another `Blocks`. Here's an example:

```py
import gradio as gr
import datetime

with gr.Blocks() as demo:
    def get_time():
        return datetime.datetime.now().time()
    dt = gr.Textbox(label="Current time", interactive=True)
    dt2 = gr.Textbox(label="Current time")
    demo.load(get_time, inputs=None, outputs=dt).then(get_time, inputs=None, outputs=dt2)

with gr.Blocks() as demo2:
    demo.render()

demo2.launch()
```

This produces an error for me. Can you see if you are able to push a fix for this?

**Comment by anentropic** [comment]
@abidlabs which error do you get?

The error I see is:
```
  File "/Users/anentropic/Documents/Dev/Sources/gradio/gradio/analytics.py", line 117, in <listcomp>
    str(blocks.blocks[y]) for y in x["targets"]
KeyError: 0
```

This can be fixed by adding a line:
```python
os.environ["GRADIO_ANALYTICS_ENABLED"] = "False"
```

(which is used in a bunch of the existing tests)

With this code I don't get an error:
```python
import datetime
import os

import gradio as gr

os.environ["GRADIO_ANALYTICS_ENABLED"] = "False"


if __name__ == "__main__":
    with gr.Blocks() as demo:

        def get_time():
            return datetime.datetime.now().time()

        dt = gr.Textbox(label="Current time", interactive=True)
        dt2 = gr.Textbox(label="Current time")
        demo.load(get_time, inputs=None, outputs=dt).then(
            get_time, inputs=None, outputs=dt2
        )

    with gr.Blocks() as demo2:
        demo.render()

    demo2.launch()
```

and the `then` chaining seems to work:
Screenshot 2023-05-24 at 14 58 11

**Comment by abidlabs** [comment]
Thanks you're right @anentropic. I relaxed the analytics function and made one other fix. Should be good to merge once CI passes.

Thanks for the contribution!

**Pull request merged**